src/dfs.c

```diff
@@ -0,0 +1,81 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "dfs.h"
+#include "Connector.h"
+
+int*** solution;
+int max_depth;
+
+/**
```

Comment:
Also rename dfs.c and dfs.h to Dfs.c and Dfs.h to follow our **naming convention**